spec/registry.md

```diff
@@ -833,3 +818,25 @@ For more information, see [Working with Timezones](https://w3c.github.io/timezon
 > The form of these serializations is known and is a de facto standard.
 > Support for these extensions is expected to be required in the post-tech preview.
 > See: https://datatracker.ietf.org/doc/draft-ietf-sedate-datetime-extended/
+
+### Date and Time Locale Options
+
+**_<dfn>Date/time locale options</dfn>_** override options set by the current locale
+or an implementation-defined date/time _operand_ value.
+
+The following options and their values are required to be available on
+the functions `:datetime`, `:date`, and `:time`:
+
+- `hour12`
+  - `true`
+  - `false`
+- `calendar` \[OPTIONAL\]
+  - valid [Unicode Calendar Identifier](https://cldr-smoke.unicode.org/spec/main/ldml/tr35.html#UnicodeCalendarIdentifier)
+- `numberingSystem` \[OPTIONAL\]
+  - valid [Unicode Number System Identifier](https://cldr-smoke.unicode.org/spec/main/ldml/tr35.html#UnicodeNumberSystemIdentifier)
+- `timeZone` \[OPTIONAL\]
+  - valid identifier per [BCP175](https://www.rfc-editor.org/rfc/rfc6557)
```

Comment:
```suggestion
  - valid identifier per either [BCP175](https://www.rfc-editor.org/rfc/rfc6557) or [Unicode Timezone Identifier](https://cldr-smoke.unicode.org/spec/main/ldml/tr35.html#UnicodeTimezoneIdentifier)
```

ICU4X and some other implementations use the short timezone IDs, so we should allow them.

Reply:
No, we shouldn't. This would effectively **require** every implementation to support them. I would rather permit them as implementation-defined.

Reply:
Here's a stab at a full fix of `timeZone`. Note that I replaced `none` with `local` for consistency with HTML and Java Temporal. I would use `float` but want to reduce cognitive burden.

```suggestion
  - well-formed identifier per [BCP175](https://www.rfc-editor.org/bcp/bcp175)
  - an implementation-defined value or identifier 
    (such as a [Unicode Timezone Identifier](https://cldr-smoke.unicode.org/spec/main/ldml/tr35.html#UnicodeTimezoneIdentifier))
  - `local`
  
> [!NOTE]
> The value `local` permits a _message_ to convert a date/time value
> into a [floating](https://www.w3.org/TR/timezone/#floating) time value
> (sometimes called a _plain_ or _local_ time value) by removing
> the association with a specific time zone.

> [!NOTE]
> Implementations SHOULD check if identifiers for each of these _options_ are valid;
> they SHOULD ignore invalid or unknown values;
> and they SHOULD emit a _Bad Option_ error for invalid or unknown values.
```

Reply:
> No, we shouldn't. This would effectively require every implementation to support them. I would rather permit them as implementation-defined.

In general, I really don't think we want to require every implementation to support **all** valid identifiers — whereby 'support' means that they are required to do something reasonable.

For example, we shouldn't require that u:locale=def "make a difference", rather than being ignored  if that language is not supported by the implementation.

With that in mind, adding (such as a [Unicode Timezone Identifier](https://cldr-smoke.unicode.org/spec/main/ldml/tr35.html#UnicodeTimezoneIdentifier)) doesn't have a cost, because people don't have to support all the values that we imply.

BTW, [BCP175](https://www.rfc-editor.org/bcp/bcp175) doesn't define either well-formedness or validity. For well-formedness, the best would be https://github.com/eggert/tz/blob/main/theory.html, but that is not very rigorous. For validity, that would be defined by scanning certain files, and looking for lines starting with Zone.

Reply:
I agree about not requiring _support_ for all valid identifiers and agree that unknown values should be ignored (but not necessarily dropped, in case they are consumed downstream).

What I'm trying to guard against is _requiring_ (or implying that it is required) implementers to build support for parsing e.g. short timezone IDs. I think we want to required that they _accept_ Olson ids (for the definition of "accept" I explained elsewhere). They are not required to do anything with any value (although users will be unhappy if real time zone IDs don't work). The formulation you're suggesting would strongly suggest that the short IDs "have" to be supported.

> BTW, [BCP175](https://www.rfc-editor.org/bcp/bcp175) doesn't define either well-formedness or validity.

Yeah, I noticed that yesterday. I hadn't been into the tzinfo docs in a while.

Reply:
I'm having some trouble understanding exactly why it's worthwhile to establish a canonical definition of a well-formed timezone identifier within the MF2 spec for an optional `timeZone` option.

This does not seem like something that's fully baked yet.

On a deeper level, I'm no longer sure that it's a good idea to forbid a datetime formatter from emitting a Bad Option error and using fallback representation when it's told to override a timezone with a value that it doesn't support, in particular as the output might not include a timezone identifier. That seems highly likely to produce misleading results.

Also, the latest suggestion above includes normative language in a non-normative note.

Reply:
Yeah, I'm not excited about defining the ABNF here in the least. I _want_ to point to CLDR, but that would bring in all of the short identifier crud.

> On a deeper level, I'm no longer sure that it's a good idea to forbid a datetime formatter from emitting a Bad Option error and using fallback representation when it's told to override a timezone with a value that it doesn't support, in particular as the output might not include a timezone identifier. That seems highly likely to produce misleading results.

I agree that we don't just want to drop the zone on the floor. You might never notice that the option is not working (or only sometimes is not working). This is particularly true for `:date` formatting (where the time zone _does_ matter, but the zone ID is almost never displayed in the results).

> Also, the latest suggestion above includes normative language in a non-normative note.

We should just get rid of that note. The problem of well-formed-but-invalid values seems like a problem for the function handler implementation. We permit errors to come out in formatting.md. Any reason to prescribe what each function does with each option?

```suggestion
  - well-formed time zone identifier
    (see [BCP175](https://www.rfc-editor.org/bpc/bpc175))
  - `local`
  - `UTC`
  
 A time zone identifier is well-formed if it matches `tzId` in the following ABNF:
>```abnf
> tzId   = tzPath / tzEtc
> tzPath = tzPart 1*("/" tzPart)
> tzPart = tzWord *("_" tzWord)
> tzWord = (%x41-5A) *(%x61-7A) ; Uppercase ASCII letter followed by lowercase letters
> tzEtc  = ("Etc/" ("UTC" / "GMT" (("+" / "-") 1*2DIGIT))
>```
  
> [!NOTE]
> The value `local` permits a _message_ to convert a date/time value
> into a [floating](https://www.w3.org/TR/timezone/#floating) time value
> (sometimes called a _plain_ or _local_ time value) 
> by removing the association with a specific time zone.
```